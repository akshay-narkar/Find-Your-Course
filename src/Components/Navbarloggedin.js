import { Link } from 'react-router-dom';

export default function Navloggedin() {
  const userdeets = JSON.parse((sessionStorage.getItem('userdetails')));
  let userid;
  let username;
  if (sessionStorage.getItem('userdetails')) {
    userid = userdeets.userid;
    username = userdeets.username;
  }

  return (
    <>
      <li>
        <Link to={{ pathname: `/users/${userid}/favs` }} className="nav-link active fw-bold" aria-current="page">
          {username}
          `s Favs
        </Link>
      </li>
      <li>
        <Link to={{ pathname: '/logout' }} className="nav-link active fw-light" aria-current="page">Logout</Link>
      </li>
    </>

  );
}
