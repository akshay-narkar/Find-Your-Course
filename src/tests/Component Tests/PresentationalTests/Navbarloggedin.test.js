/*eslint-disable*/
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Navbarloggedin from '../../../Components/Navbarloggedin';
import store from '../store';

it('UI is unaltered wrt first snapshot', () => {
  const elem = renderer.create(<BrowserRouter><Provider store={store}><Navbarloggedin /></Provider></BrowserRouter>).toJSON();

  expect(elem).toMatchSnapshot();
});
