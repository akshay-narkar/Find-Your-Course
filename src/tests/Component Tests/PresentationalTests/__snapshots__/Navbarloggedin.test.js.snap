// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI is unaltered wrt first snapshot 1`] = `
Array [
  <li>
    <a
      aria-current="page"
      className="nav-link active fw-bold"
      href="/users/null/favs"
      onClick={[Function]}
    >
      \`s Favs
    </a>
  </li>,
  <li>
    <a
      aria-current="page"
      className="nav-link active fw-light"
      href="/logout"
      onClick={[Function]}
    >
      Logout
    </a>
  </li>,
]
`;
